src/evergreen_mcp/server.py

```diff
@@ -109,163 +121,112 @@ async def _server_lifespan(_) -> AsyncIterator[dict]:
-        }
-        return [
-            types.TextContent(type="text", text=json.dumps(error_response, indent=2))
-        ]
 
-    # Call the appropriate handler
-    try:
-        logger.debug("Executing tool: %s", name)
+# Import and register tools after mcp is created
+from evergreen_mcp.tools import register_tools  # noqa: E402
 
-        # Use DEFAULT_PROJECT_ID as fallback if not provided in arguments
-        if "project_id" not in arguments and DEFAULT_PROJECT_ID:
-            arguments = {**arguments, "project_id": DEFAULT_PROJECT_ID}
-            logger.info("Using default project ID: %s", DEFAULT_PROJECT_ID)
+register_tools(mcp)
 
-        if name == "list_user_recent_patches_evergreen":
-            result = await handler(arguments, client, USER_ID)
-        else:
-            result = await handler(arguments, client)
-        logger.debug("Tool %s completed successfully", name)
-        return result
-    except Exception as e:
-        logger.error("Tool handler failed for %s", name, exc_info=True)
-        error_response = {
-            "error": f"Tool execution failed: {str(e)}",
-            "tool": name,
-            # Removed arguments to avoid logging potentially sensitive data
-        }
-        return [
-            types.TextContent(type="text", text=json.dumps(error_response, indent=2))
-        ]
-
-
-async def _main() -> int:
-    logger.info("Setting up MCP stdio server...")
-    async with mcp.server.stdio.stdio_server() as (read_stream, write_stream):
-        logger.info("MCP server running and ready for connections")
-        await server.run(
-            read_stream,
-            write_stream,
-            InitializationOptions(
-                server_name="evergreen-mcp-server",
-                server_version="0.1.0",
-                capabilities=server.get_capabilities(
-                    notification_options=NotificationOptions(),
-                    experimental_capabilities={},
-                ),
-            ),
+
+# Register resources
+@mcp.resource("evergreen://projects")
+async def list_projects_resource(ctx: Context) -> str:
+    """List all Evergreen projects as a resource."""
+    import json
+
+    evg_ctx = ctx.request_context.lifespan_context
+    try:
+        projects = await evg_ctx.client.get_projects()
+        return json.dumps(
+            [
+                {
+                    "id": p.get("id"),
+                    "identifier": p.get("identifier"),
+                    "displayName": p.get("displayName"),
+                    "enabled": p.get("enabled"),
+                    "owner": p.get("owner"),
+                    "repo": p.get("repo"),
+                }
+                for p in projects
+            ],
+            indent=2,
         )
-    logger.info("MCP server shutting down")
-    return 0
+    except Exception as e:
+        logger.error("Failed to fetch projects resource: %s", e)
+        return json.dumps({"error": str(e)})
```

Comment:
How does FastMCP do error handling if a tool raises an exception? I'm wondering if we need to do any special catching of exceptions or if we can rely on FastMCP to return the error response unless we need some custom handling.